### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by laugengebaeck**
## What does this change?

This adds missing tests for classes with poor test coverage. Currently finished are:

- [x] `ICAccountInfo`
- [x] `ICConnectionHandler`
- [x] `ICEmail`
- [x] `ICEndPoint`
- [x] `ICFolder`
- [x] `ICStreamWrapper` (wrapper class with only non-testable methods left)

Co-authored-by: Smilla Fox <foxsmilla@gmail.com>
Co-authored-by: Lukas Rost <lukas.rost@student.hpi.de>

## Checklist

- [x] Your branch has no merge conflicts with master
- [x] All related commits are squashed together

### CODE REVIEW

File: packages/IMAPClient-Tests.package/ICEndPointAuthentificationTest.class/instance/setUp.st [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+		responses: self endPoint stream server notAuthenticatedResponses.
+		
+	accountInfo := ((ICAccountInfo new)
+		accountName:  'testAccountName';
+		host: 'test.host';
+		port: 993;
+		username: 'testUser';
+		password: 'superSecure';
+		ssl: true).
+
```
Comment (elenagensch) [inline packages/IMAPClient-Tests.package/ICEndPointAuthentificationTest.class/instance/setUp.st]: accountInfo makes the mehtod long, could be extracted in a (constant) method.

**Review by elenagensch — APPROVED** [review]

**Comment by laugengebaeck** [comment]
I found a really weird code duplication in `ICFolder>>retrieveBodyFor:` and `ICEmail>>retrieveBody`. Which one is the "right" one and which one shoud be tested?

Does someone know more about this (@jannikgro or @elenagensch maybe)?

Edit: Jan said the `ICFolder` one was the right one and the other is dead code (now removed)

**Comment by laugengebaeck** [comment]
Most of the remaining untested methods in `ICConnectionHandler`, `ICEndpoint` and `ICFolder` cover server interaction. We already have an `ICMockServer` which we could use for that, but it seems complicated to implement. As our focus isn't TDD, we should probably stop here. What do you think?

**Comment by sm1lla** [comment]
I agree. These aren't methods we implemented ourself and it would take too much time to understand `ICMockServer` as we haven't worked intensively with the IMAP protocol yet. Let's leave the remaining methods untested for now and come back to them if we have more time in another sprint.

File: packages/IMAPClient-Tests.package/ICMockReadWriter.class/instance/delete..st [hunk 2]
```diff
@@ -0,0 +1,3 @@
+mockMethods
+delete: anArgument
+
```
Comment (elenagensch) [inline packages/IMAPClient-Tests.package/ICMockReadWriter.class/instance/delete..st]: remove newline

**Review by elenagensch — APPROVED** [review]

**Comment by laugengebaeck** [comment]
Would also merge this today (maybe somebody wants to do a quick review again :)

**Review by blackjack2693 — APPROVED** [review]

**Pull request merged**